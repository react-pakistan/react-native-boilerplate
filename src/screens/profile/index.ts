export * from './profile';
export * from './type';
