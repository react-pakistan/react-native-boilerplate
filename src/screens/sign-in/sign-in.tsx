import { Button, Spacer } from '@react-pakistan/react-native-commons-collection';
import React, { ReactElement, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { useTheme } from 'styled-components';
import { showApp } from '../../redux/slices/main-slice';
import { loginScreenText } from './helpers';
import {
  OAuthHeading,
  SignInWrapper,
} from './styled';

export const SignIn = () : ReactElement => {
  // dispatch
  const dispatch = useDispatch();

  // theme
  const theme = useTheme();

  // callbacks
  const onPressHandler = useCallback(() : void => {
    dispatch(showApp());
  }, [dispatch]);

  return (
    <SignInWrapper>
      <OAuthHeading>
        {loginScreenText.oAuthHeading}
      </OAuthHeading>
      <Spacer marginVertical={theme.spacing.xsmall} />
      <Button
        onPress={onPressHandler}
        title={loginScreenText.facebookButton}
      />
      <Spacer marginVertical={theme.spacing.xsmall} />
      <Button
        onPress={onPressHandler}
        title={loginScreenText.googleButton}
      />
    </SignInWrapper>
  );
};
