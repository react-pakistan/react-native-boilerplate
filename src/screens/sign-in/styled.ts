import { IRNTheme } from '@react-pakistan/util-react-native-functions';
import { Text, View } from '@react-pakistan/react-native-commons-collection';
import { ThemedStyledProps } from 'styled-components';
import styled from 'styled-components/native';

export const SignInWrapper = styled(View)`
  align-items: center;
  background-color: ${({
    theme,
  } : ThemedStyledProps<unknown, IRNTheme>) : string => theme.colors.lightGrey};
  flex: 1;
  justify-content: center;
`;

export const OAuthHeading = styled(Text)`
  color: ${({
    theme,
  } : ThemedStyledProps<unknown, IRNTheme>) : string => theme.colors.black};
  font-family: ${({
    theme,
  } : ThemedStyledProps<unknown, IRNTheme>) : string => theme.typography.fontFamily};
  font-size: ${({
    theme,
  } : ThemedStyledProps<unknown, IRNTheme>) : number => theme.spacing.default}px;
  text-transform: uppercase;
`;
