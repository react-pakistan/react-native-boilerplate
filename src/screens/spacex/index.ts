export * from './spacex';
export * from './type';
