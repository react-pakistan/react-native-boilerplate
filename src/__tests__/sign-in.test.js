import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { SignIn } from '../screens/sign-in';

const fakeInitialState = {
  showAppReducer: {
    showApp: true,
  },
};

const mockStore = configureStore();

test('renders SignIn component correctly', () => {
  const renderer = new ShallowRenderer();
  const showApp = false;
  const store = mockStore(fakeInitialState);
  const showAppAction = () => null;
  renderer.render(
    <Provider store={store}>
      <SignIn showApp={showApp} showAppAction={showAppAction} />
    </Provider>,
  );
  const wrapper = renderer.getRenderOutput();
  expect(wrapper).toMatchSnapshot();
});
