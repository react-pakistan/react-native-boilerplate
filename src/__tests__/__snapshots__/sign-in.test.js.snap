// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SignIn component correctly 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <SignIn
    showApp={false}
    showAppAction={[Function]}
  />
</Context.Provider>
`;
